In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.neighbors import KNeighborsClassifier

In [2]:
df = pd.read_csv("ai_student_impact_dataset.csv")
df.head()
df = df.drop("Student_ID", axis=1)
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
0,Humanities,Senior,2.418,23.31,Copywriting/Drafting,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High
1,Medical,Junior,3.821,1.12,Ideation,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low
2,Business,Freshman,3.398,21.26,Summarizing_Reading,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium
3,Business,Senior,3.789,1.82,Copywriting/Drafting,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium
4,STEM,Sophomore,3.635,9.29,Debugging/Troubleshooting,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium


In [3]:
print(df["Major_Category"].unique())

<StringArray>
['Humanities', 'Medical', 'Business', 'STEM', 'Arts']
Length: 5, dtype: str


In [4]:
df["Major_Category"] = df["Major_Category"].map({
    "Humanities":1,
    "Medical":2,
    "Business":3,
    "STEM":4,
    "Arts":5
})
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
0,1,Senior,2.418,23.31,Copywriting/Drafting,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High
1,2,Junior,3.821,1.12,Ideation,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low
2,3,Freshman,3.398,21.26,Summarizing_Reading,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium
3,3,Senior,3.789,1.82,Copywriting/Drafting,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium
4,4,Sophomore,3.635,9.29,Debugging/Troubleshooting,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium


In [5]:
print(df["Year_of_Study"].unique())

<StringArray>
['Senior', 'Junior', 'Freshman', 'Sophomore', 'Graduate']
Length: 5, dtype: str


In [6]:
df["Year_of_Study"] = df["Year_of_Study"].map({
    "Senior":1,
    "Junior":2,
    "Freshman":3,
    "Sophomore":4,
    "Graduate":5
})
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
0,1,1,2.418,23.31,Copywriting/Drafting,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High
1,2,2,3.821,1.12,Ideation,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low
2,3,3,3.398,21.26,Summarizing_Reading,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium
3,3,1,3.789,1.82,Copywriting/Drafting,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium
4,4,4,3.635,9.29,Debugging/Troubleshooting,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium


In [7]:
df = pd.get_dummies(df,columns=["Primary_Use_Case"],dtype=int)
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level,Primary_Use_Case_Copywriting/Drafting,Primary_Use_Case_Debugging/Troubleshooting,Primary_Use_Case_Direct_Answer_Generation,Primary_Use_Case_Ideation,Primary_Use_Case_Summarizing_Reading
0,1,1,2.418,23.31,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High,1,0,0,0,0
1,2,2,3.821,1.12,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low,0,0,0,1,0
2,3,3,3.398,21.26,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium,0,0,0,0,1
3,3,1,3.789,1.82,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium,1,0,0,0,0
4,4,4,3.635,9.29,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium,0,1,0,0,0


In [8]:
print(df["Prompt_Engineering_Skill"].unique())

<StringArray>
['Beginner', 'Advanced', 'Intermediate']
Length: 3, dtype: str


In [9]:
df["Prompt_Engineering_Skill"] = df["Prompt_Engineering_Skill"].map({
    "Beginner":1,
    "Intermediate":2,
    "Advanced":3
})
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level,Primary_Use_Case_Copywriting/Drafting,Primary_Use_Case_Debugging/Troubleshooting,Primary_Use_Case_Direct_Answer_Generation,Primary_Use_Case_Ideation,Primary_Use_Case_Summarizing_Reading
0,1,1,2.418,23.31,1,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High,1,0,0,0,0
1,2,2,3.821,1.12,3,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low,0,0,0,1,0
2,3,3,3.398,21.26,1,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium,0,0,0,0,1
3,3,1,3.789,1.82,2,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium,1,0,0,0,0
4,4,4,3.635,9.29,3,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium,0,1,0,0,0


In [10]:
df["Paid_Subscription"] = df["Paid_Subscription"].map({
    True:1,
    False:0
})
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level,Primary_Use_Case_Copywriting/Drafting,Primary_Use_Case_Debugging/Troubleshooting,Primary_Use_Case_Direct_Answer_Generation,Primary_Use_Case_Ideation,Primary_Use_Case_Summarizing_Reading
0,1,1,2.418,23.31,1,1,1,8.13,5,Allowed_With_Citation,6,2.393,86.44,High,1,0,0,0,0
1,2,2,3.821,1.12,3,5,0,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low,0,0,0,1,0
2,3,3,3.398,21.26,1,2,0,10.35,5,Strict_Ban,9,3.499,73.93,Medium,0,0,0,0,1
3,3,1,3.789,1.82,2,4,0,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium,1,0,0,0,0
4,4,4,3.635,9.29,3,4,0,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium,0,1,0,0,0


In [11]:
print(df["Institutional_Policy"].unique())

<StringArray>
['Allowed_With_Citation', 'Strict_Ban', 'Actively_Encouraged']
Length: 3, dtype: str


In [12]:
df["Institutional_Policy"] = df["Institutional_Policy"].map({
    "Allowed_With_Citation":2,
    "Strict_Ban":0,
    "Actively_Encouraged":1
})
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level,Primary_Use_Case_Copywriting/Drafting,Primary_Use_Case_Debugging/Troubleshooting,Primary_Use_Case_Direct_Answer_Generation,Primary_Use_Case_Ideation,Primary_Use_Case_Summarizing_Reading
0,1,1,2.418,23.31,1,1,1,8.13,5,2,6,2.393,86.44,High,1,0,0,0,0
1,2,2,3.821,1.12,3,5,0,16.65,3,2,9,3.696,69.39,Low,0,0,0,1,0
2,3,3,3.398,21.26,1,2,0,10.35,5,0,9,3.499,73.93,Medium,0,0,0,0,1
3,3,1,3.789,1.82,2,4,0,15.23,2,2,2,4.000,63.58,Medium,1,0,0,0,0
4,4,4,3.635,9.29,3,4,0,12.55,4,2,4,3.798,100.00,Medium,0,1,0,0,0


In [13]:
print(df["Burnout_Risk_Level"].unique())

<StringArray>
['High', 'Low', 'Medium']
Length: 3, dtype: str


In [14]:
df["Burnout_Risk_Level"] = df["Burnout_Risk_Level"].map({
    "High":2,
    "Low":0,
    "Medium":1
})
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level,Primary_Use_Case_Copywriting/Drafting,Primary_Use_Case_Debugging/Troubleshooting,Primary_Use_Case_Direct_Answer_Generation,Primary_Use_Case_Ideation,Primary_Use_Case_Summarizing_Reading
0,1,1,2.418,23.31,1,1,1,8.13,5,2,6,2.393,86.44,2,1,0,0,0,0
1,2,2,3.821,1.12,3,5,0,16.65,3,2,9,3.696,69.39,0,0,0,0,1,0
2,3,3,3.398,21.26,1,2,0,10.35,5,0,9,3.499,73.93,1,0,0,0,0,1
3,3,1,3.789,1.82,2,4,0,15.23,2,2,2,4.000,63.58,1,1,0,0,0,0
4,4,4,3.635,9.29,3,4,0,12.55,4,2,4,3.798,100.00,1,0,1,0,0,0


In [15]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 19 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Major_Category                              50000 non-null  int64  
 1   Year_of_Study                               50000 non-null  int64  
 2   Pre_Semester_GPA                            50000 non-null  float64
 3   Weekly_GenAI_Hours                          50000 non-null  float64
 4   Prompt_Engineering_Skill                    50000 non-null  int64  
 5   Tool_Diversity                              50000 non-null  int64  
 6   Paid_Subscription                           50000 non-null  int64  
 7   Traditional_Study_Hours                     50000 non-null  float64
 8   Perceived_AI_Dependency                     50000 non-null  int64  
 9   Institutional_Policy                        50000 non-null  int64  
 10  Anxiety_Level_During_

In [16]:
df.head()

,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level,Primary_Use_Case_Copywriting/Drafting,Primary_Use_Case_Debugging/Troubleshooting,Primary_Use_Case_Direct_Answer_Generation,Primary_Use_Case_Ideation,Primary_Use_Case_Summarizing_Reading
0,1,1,2.418,23.31,1,1,1,8.13,5,2,6,2.393,86.44,2,1,0,0,0,0
1,2,2,3.821,1.12,3,5,0,16.65,3,2,9,3.696,69.39,0,0,0,0,1,0
2,3,3,3.398,21.26,1,2,0,10.35,5,0,9,3.499,73.93,1,0,0,0,0,1
3,3,1,3.789,1.82,2,4,0,15.23,2,2,2,4.000,63.58,1,1,0,0,0,0
4,4,4,3.635,9.29,3,4,0,12.55,4,2,4,3.798,100.00,1,0,1,0,0,0


In [ ]:
print(df.corr(numeric_only=True)["Burnout_Risk_Level"].sort_values(ascending=False))

Burnout_Risk_Level                            1.000000
Weekly_GenAI_Hours                            0.469472
Perceived_AI_Dependency                       0.369318
Anxiety_Level_During_Exams                    0.164505
Paid_Subscription                             0.103783
Major_Category                                0.054987
Year_of_Study                                 0.032961
Primary_Use_Case_Debugging/Troubleshooting    0.029905
Primary_Use_Case_Direct_Answer_Generation     0.012585
Prompt_Engineering_Skill                      0.009686
Tool_Diversity                                0.005783
Primary_Use_Case_Ideation                    -0.006359
Primary_Use_Case_Summarizing_Reading         -0.011465
Primary_Use_Case_Copywriting/Drafting        -0.023693
Institutional_Policy                         -0.035849
Skill_Retention_Score                        -0.058140
Post_Semester_GPA                            -0.095543
Pre_Semester_GPA                             -0.096234
Traditiona

Burnout risk is most strongly associated with Weekly GenAI Usage Hours (r = 0.469) and Perceived AI Dependency (r = 0.369). Traditional Study Hours, Pre-Semester GPA, and Post-Semester GPA showed weak negative correlations with burnout risk, suggesting that increased study habits and academic performance were associated with slightly lower burnout levels.

In [18]:
from sklearn.metrics import accuracy_score
from sklearn.ensemble import RandomForestClassifier

X = df.drop("Burnout_Risk_Level", axis=1)
y = df["Burnout_Risk_Level"]

X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2,random_state=42)


model = RandomForestClassifier(n_estimators=200)
model.fit(X_train,y_train)

prediction = model.predict(X_test)
print("Actual")
print(y_test[:10])
print("Predicted")
print(prediction[:10])
print(accuracy_score(y_test,prediction))

Actual
33553    0
9427     0
199      2
12447    2
39489    0
42724    0
10822    2
49498    2
4144     0
36958    0
Name: Burnout_Risk_Level, dtype: int64
Predicted
[2 1 2 1 1 0 1 1 1 1]
0.5277


In [19]:
print(df["Burnout_Risk_Level"].value_counts())

Burnout_Risk_Level
1    21144
0    16369
2    12487
Name: count, dtype: int64


In [20]:
print(y_train.value_counts())

Burnout_Risk_Level
1    16949
0    13100
2     9951
Name: count, dtype: int64
